jupyter-extension/src/plugins/mount/api.ts

```diff
@@ -0,0 +1,82 @@
+import {Repo, Mount, ListMountsResponse} from './types';
+import {requestAPI} from '../../handler';
+
+export const unmountAll = async (
+  updateData?: (response: ListMountsResponse) => void,
+): Promise<void> => {
+  const response = await requestAPI<ListMountsResponse>('_unmount_all', 'PUT');
+  if (updateData) {
+    return updateData(response);
+  }
+
+  return;
+};
+
+export const mount = async (
+  updateData: (response: ListMountsResponse) => void,
+  projectRepo: string,
+  branch = 'master',
+): Promise<void> => {
+  const [project, repo] = projectRepo.split('/');
+
+  await unmountAll();
```

Comment:
i think there's an opportunity to simplify the backend as well as remove this logic. with the assumption that only one branch is mounted at a time, things get a lot simpler. we can discuss doing this in a later revision if need be.

Reply:
Good point, let me take a look at what would be involved in doing that. If not in this PR definitely a follow up right after.

Reply:
happy to help walk you through the file manager code in the backend when you are ready to do this!